src/marqo/core/unstructured_vespa_index/unstructured_document.py

```diff
@@ -175,8 +187,18 @@ def to_marqo_document(self, return_highlights: bool = False) -> Dict[str, Any]:
                     json.loads(serialized_multimodal_params)
 
         if return_highlights and self.fields.match_features:
-            if not self.fields.vespa_chunks:
-                raise VespaDocumentParsingError(f"Document {self.fields.marqo__id} does not have any chunks. "
-                                                f"No highlights can be extracted.")
-            marqo_document[index_constants.MARQO_DOC_HIGHLIGHTS] = self.fields.extract_highlights()
+            if self.fields.match_features[f"closest({unstructured_common.VESPA_DOC_EMBEDDINGS})"]["cells"] == {}:
```

Comment:
How did this code handle lexical before this change?

Reply:
There would be no `self.fields.match_features` in that case, because it's only in the embedding similarity rank profile. This issue arises because there's a possibility now for a doc with match features (but it's empty) and empty vespa chunks.

Reply:
As agreed move this into `self.fields.extract_highlights`